src/test/java/com/techsophy/tsf/account/service/ACLServiceTest.java

```diff
@@ -0,0 +1,232 @@
+import static com.techsophy.tsf.account.constants.AccountConstants.EMAIL_ID;
+import static com.techsophy.tsf.account.constants.AccountConstants.FIRST_NAME;
+import static com.techsophy.tsf.account.constants.AccountConstants.ID;
+import static com.techsophy.tsf.account.constants.AccountConstants.LAST_NAME;
+import static com.techsophy.tsf.account.constants.AccountConstants.UPDATED_BY_ID;
+import static com.techsophy.tsf.account.constants.AccountConstants.UPDATED_BY_NAME;
+import static com.techsophy.tsf.account.constants.AccountConstants.UPDATED_ON;
+import static com.techsophy.tsf.account.constants.AccountConstants.*;
+import static com.techsophy.tsf.account.constants.ThemesConstants.*;
+import static com.techsophy.tsf.account.constants.UserPreferencesConstants.DEPARTMENT;
+import static com.techsophy.tsf.account.constants.UserPreferencesConstants.MOBILE_NUMBER;
+import static com.techsophy.tsf.account.constants.UserPreferencesConstants.NULL;
+import static com.techsophy.tsf.account.constants.UserPreferencesConstants.NUMBER;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ACLServiceTest
+{
+    @Mock
+    GlobalMessageSource globalMessageSource;
+    @Mock
+    IdGeneratorImpl mockIdGenerator;
+    @Mock
+    TokenUtils mockTokenUtils;
+    @Mock
+    UserFormDataDefinitionRepository userFormDataDefinitionRepository;
+    @Mock
+    ObjectMapper mockObjectMapper;
+    @Mock
+    UserDetails mockUserDetails;
+    @Mock
+    ACLRepository aclRepository;
+    @InjectMocks
+    ACLServiceImpl aclService;
+
+    List<Map<String, Object>> userList = new ArrayList<>();
+
+    @BeforeEach
+    public void init()
+    {
+        Map<String, Object> map = new HashMap<>();
+        map.put(CREATED_BY_ID, NULL);
+        map.put(CREATED_BY_NAME, NULL);
+        map.put(CREATED_ON, NULL);
+        map.put(UPDATED_BY_ID, NULL);
+        map.put(UPDATED_BY_NAME, NULL);
+        map.put(UPDATED_ON, NULL);
+        map.put(ID, BIGINTEGER_ID);
+        map.put(USER_NAME_DATA, USER_FIRST_NAME);
+        map.put(FIRST_NAME, USER_LAST_NAME);
+        map.put(LAST_NAME, USER_FIRST_NAME);
+        map.put(MOBILE_NUMBER, NUMBER);
+        map.put(EMAIL_ID, MAIL_ID);
+        map.put(DEPARTMENT, NULL);
+        userList.add(map);
+    }
+
+    @Test
+    void saveACLFreshRecordTest() throws JsonProcessingException
+    {
+        Mockito.when(mockUserDetails.getUserDetails())
+                .thenReturn(userList);
+        ACLSchema aclSchema=new ACLSchema();
+        when(mockIdGenerator.nextId()).thenReturn(BigInteger.valueOf(Long.parseLong(BIGINTEGER_ID)));
+        ACLDefinition aclDefinition=new ACLDefinition();
+        Mockito.when(mockObjectMapper.convertValue(any(), ArgumentMatchers.eq(ACLDefinition.class))).thenReturn(aclDefinition);
+        Assertions.assertNotNull(aclService.saveACL(aclSchema));
```

Comment:
Please add test cases for failure conditions, what if the name is invalid or already existing?
 or what if mongo connection is not avialable